libs/payments/cart/src/lib/manager.ts

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { v4 as uuidv4 } from 'uuid';
+import { Cart } from '../../../../shared/db/mysql/account/src';
+import { uuidTransformer } from '../../../../shared/db/mysql/core/src';
+import { Logger } from '../../../../shared/log/src';
+import { Cart as CartType } from '../gql';
+import { UpdateCartInput } from '../gql/update-cart.input';
+import { InvoiceFactory } from './factories';
+import { CartState, SetupCart } from './types';
+
+const DEFAULT_INTERVAL = 'monthly';
+
+export class CartManager {
+  private log: Logger;
+  constructor(log: Logger) {
+    this.log = log;
+  }
+
+  public async setupCart(input: SetupCart): Promise<CartType | null> {
+    const id = uuidv4({}, Buffer.alloc(16)).toString('hex');
+    const currentDate = Date.now();
+    const cart = await Cart.query().insert({
+      ...input,
+      id,
+      state: CartState.START,
+      interval: input.interval || DEFAULT_INTERVAL,
+      createdAt: currentDate,
+      updatedAt: currentDate,
+    });
+
+    return {
+      ...cart,
+      nextInvoice: InvoiceFactory(), // Temporary
+    };
+  }
+
+  public async restartCart(cartId: string): Promise<CartType | null> {
+    const id = uuidTransformer.to(cartId);
+    const cart = await Cart.query().where('id', id).first();
+    if (!cart) {
+      return null;
```

Comment:
If no cart is found with the ID provided, we should probably throw an error.

Reply:
That makes sense. I'm adding on `.throwIfNotFound()`.